### PR TITLE
1967 create proper signatures for argument structures

### PR DESCRIPTION
**Pull request by hiker**
A small patch so that arguments that are structures (`something%ncell_3d`) are turned into proper signatures when collecting the variable accesses. Previously it would be using a 1-element using the whole string, instead of a 2-element signature with `(something, ncell_3d)`.

### CODE REVIEW

**Comment by hiker** [comment]
Ready for review, the link check failures appear to be unrelated.

**Review by arporter — CHANGES_REQUESTED** [review]
Looks good.
All tests and examples are OK with compilation.
Ref. guide builds without new warnings.
Just a couple of minor things to address.

File: src/psyclone/core/signature.py [hunk 1]
```diff
@@ -37,8 +37,6 @@
 
 '''This module provides management of variable access information.'''
```
[the remark this answers is not in the document]
Comment (hiker) [reply src/psyclone/core/signature.py:39]: Done.

File: src/psyclone/tests/domain/lfric/kern_call_arg_list_test.py [hunk 2]
```diff
@@ -352,7 +352,8 @@ def test_kerncallarglist_bcs_operator(fortran_writer):
 
     schedule = psy.invokes.invoke_list[0].schedule
     create_arg_list = KernCallArgList(schedule.kernels()[0])
-    create_arg_list.generate()
+    access_info = VariablesAccessInfo()
+    create_arg_list.generate(access_info)
     assert create_arg_list._arglist == [
         'cell', 'nlayers', 'op_a_proxy%ncell_3d', 'op_a_proxy%local_stencil',
```
[the remark this answers is not in the document]
Comment (hiker) [reply src/psyclone/tests/domain/lfric/kern_call_arg_list_test.py:358]: @arporter I am not sure how to create this kind of code? As far as I can see, `op_a_proxy(2)` is in the PSyLayer, i.e. must be created by PSyclone. Even if I declare an array:
```
  type(operator_type) :: op_a(3)  
  call invoke(enforce_operator_bc_kernel_type(op_a(2)))
```
the PSyLayer will just see a single `operator_type` argument, so it won't create an array of proxies. I tried to create a vector of operators, but got:
```
In the LFRic API, vector notation is only supported for ['gh_field'] argument types but found 'gh_operator * 3'
```

So I am not sure what to do.

FWIW, vector fields are tested, e.g.:
```
    assert create_arg_list._arglist == [
        'nlayers', 'f1_proxy%data', 'f2_proxy(1)%data', 'f2_proxy(2)%data',
        'f2_proxy(3)%data', 'f3_proxy%data', 'istp', 'ndf_w2', 'undf_w2',
        'map_w2(:,cell)', 'basis_w2_qr_xyoz', 'basis_w2_qr_face', 'ndf_wchi',
```

**Comment by hiker** [comment]
Should be ready for next review.

File: src/psyclone/tests/domain/lfric/kern_call_arg_list_test.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (arporter) [reply src/psyclone/tests/domain/lfric/kern_call_arg_list_test.py:358]: Ah, that's OK then. I just didn't look very carefully - apologies.

**Review by arporter — APPROVED** [review]
All requested changes have been made.
Tests, examples and ref. guide are still fine.
Issue no. is not mentioned anywhere.
Will proceed to merge.

**Pull request merged**